### PR TITLE
[ACCSEC-22363] Make setClearStorageOnReinstall public and add documentation

### PR DESCRIPTION
**Pull request by yafuquen**
## Jira Ticket
- ACCSEC-22363

## Description
- Make setClearStorageOnReinstall public 
- Add documentation for reinstall in readme

## Commit message
-  feat: Allow persisting factors after reinstall

## Testing
- [ ] Added unit tests
- [x] Ran unit tests successfully
- [x] Added documentation for public APIs and/or Wiki

### CODE REVIEW

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.14%`**
File | Coverage
-----|-----
TwilioVerifySDK/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`84.62%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: TwilioVerifySDK/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift [hunk 1]
```diff
@@ -179,7 +179,9 @@ public class TwilioVerifyBuilder {
     return self
   }
   
-  func setClearStorageOnReinstall(_ clearStorageOnReinstall: Bool) -> Self {
+  /// Defines if the storage will be cleared after a reinstall
+  /// - Parameter clearStorageOnReinstall: If true, the storage will be cleared after a reinstall, so created factors will not exist in the device anymore. If false, created factors will persist in the device. Default value is true
```
Comment (AuthyApps) [inline TwilioVerifySDK/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift:183]: <table data-meta="generated_by_UnitTests">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/twilio/twilio-verify-ios/blob/6cb8fbf27f8bc164af4278ce9c57e362b1354c81/TwilioVerifySDK/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift#L183"></span>Line should be 195 characters or less: currently 231 characters<br />
<code>line_length</code> <code>TwilioVerify.swift:183</code></td>
    </tr>
  </tbody>
</table>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.14%`**
File | Coverage
-----|-----
TwilioVerifySDK/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`84.62%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.14%`**
File | Coverage
-----|-----
TwilioVerifySDK/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`84.62%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by yafuquen** [comment]
> Change "as it is explained in" to "as is explained in"

@TimGu Updated the README with your suggestions

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.14%`**
File | Coverage
-----|-----
TwilioVerifySDK/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`84.62%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by AlejandroOrozco — APPROVED** [review]
LG 👍

**Pull request merged**